Simplified/NYPLBookDetailViewController.m

```diff
@@ -193,8 +193,10 @@ - (void)problemReportViewController:(NYPLProblemReportViewController *)problemRe
 {
   NSURL *reportURL = problemReportViewController.book.reportURL;
   if (reportURL) {
-    NSURLRequest *r = [NSURLRequest postRequestWithProblemDocument:@{@"type":type} url:reportURL];
-    [[NYPLSession sharedSession] uploadWithRequest:r completionHandler:nil];
+    NSURLRequest *request = [NSURLRequest
+                             postRequestWithProblemDocument:@{@"type":type}
+                             url:reportURL];
+    [NYPLNetworkExecutor.shared POST:request completion:nil];
```

Comment:
as we discussed in private, can you create a new executor instance in the initializer  and save that as a "private" property? The `shared` instance is just a convenience (mainly to reuse the same caching storage) but here it makes sense to use `.ephemeral` caching:
```
let x = NYPLNetworkExecutor(cachingStrategy: .ephemeral)
```

Reply:
Sorry I forgot about doing that. Done.